### PR TITLE
Inbox Implementation

### PR DESCRIPTION
**Pull request by talin2412**

### CODE REVIEW

File: backend/src/main/kotlin/clarity/backend/entity/InboxEntity.kt [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    }
+
+    fun createNotification(notification: CreateNotification) : NotificationResponse {
+        val db = DataManager.conn()
+        try {
+            val statement = db!!.createStatement()
+            val insertStatement = """
+                INSERT INTO Inbox (user_id, message, notification_date, message_read)
+                VALUES(
+                '${notification.userId}', '${notification.message}', '${notification.notificationDate}', '${0}'
```
Comment (ad-world) [inline backend/src/main/kotlin/clarity/backend/entity/InboxEntity.kt:80]: should these integer values be wrapped in ''? will that convert them into strings in SQL or are they still integers? The same goes for all other SQL queries you've written. could you j check your database and make sure they're stored as integers and not strings?

**Review by ad-world — COMMENTED** [review]
left one comment

File: backend/src/main/kotlin/clarity/backend/entity/InboxEntity.kt (hunk 1 above)
Comment (talin2412) [reply backend/src/main/kotlin/clarity/backend/entity/InboxEntity.kt]: good catch, whats weird is that it caused no integrity constraints in any of the queries while inserting or selecting anything. I've changed all my files to match everything though, so should be good!

File: backend/src/main/kotlin/clarity/backend/entity/InboxEntity.kt [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+            return NotificationResponse(StatusResponse.Failure, "Could not mark message")
+        }
+    }
+
+    fun deleteMessage(notificationId: Int) : NotificationResponse {
+        val db = DataManager.conn()
+        try {
+            val statement = db!!.createStatement()
+            val deleteStatement = """
+                DELETE FROM Inbox WHERE user_id=${notificationId}
```
Comment (ad-world) [inline backend/src/main/kotlin/clarity/backend/entity/InboxEntity.kt:63]: is this supposed to be user_id = notificationId? or should it be notification id

File: backend/src/main/kotlin/clarity/backend/entity/InboxEntity.kt [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+
+class InboxEntity {
+
+    fun getUnread(userId: Int) : GetUnreadResponse {
+        try {
+            val db = DataManager.conn()
+            val statement = db!!.createStatement()
+            val selectStatement = """
+                SELECT * FROM Inbox WHERE user_id='${userId}' AND message_read='${0}'
```
Comment (ad-world) [inline backend/src/main/kotlin/clarity/backend/entity/InboxEntity.kt:20]:
```suggestion
                SELECT * FROM Inbox WHERE user_id='${userId}' AND message_read=0
```

**Review by ad-world — APPROVED** [review]

File: backend/src/main/kotlin/clarity/backend/entity/InboxEntity.kt (hunk 1 above)
Comment (SaumyaKhati) [reply backend/src/main/kotlin/clarity/backend/entity/InboxEntity.kt]: See Section 3 in https://www.sqlite.org/datatype3.html. It looks like it tries to cast it if you pass in a string value for the int column(s). So if it's guaranteed to be integer, then we don't have to worry about it.

File: android/app/src/main/java/com/example/clarity/sdk/ClaritySDK.kt [hunk 4]
```diff
@@ -148,6 +148,15 @@ interface API {
     @POST("toggleCardSetVisibility")
     suspend fun toggleCardSetVisibility(@Body request: ToggleCardSetRequest): Response<ToggleCardSetResponse>
 
+    @GET("getWords")
+    suspend fun getWords(): Response<GetWordsResponse>
+
+//    @POST("getProgressForSet")
```
Comment (ArshpreetChabbewal) [inline android/app/src/main/java/com/example/clarity/sdk/ClaritySDK.kt:154]: What was our reasoning for removing these endpoints again, are we depreciating set progress in general, or is this being transferred to a different endpoint.

**Pull request merged**